services/galley/src/Galley/Data/Types.hs

```diff
@@ -80,6 +83,24 @@ isConvDeleted = fromMaybe False . convDeleted
 selfConv :: UserId -> ConvId
 selfConv uid = Id (toUUID uid)
 
+--------------------------------------------------------------------------------
+
+-- | Internal teamMemberList type, corresponding to some state in Cassandra - used
+-- mostly to avoid to boolean blindness until the very end (client facing API)
+-- Should never be sent to users (and therefore doesn't have 'FromJSON' or
+-- 'ToJSON' instances).
+data ListType
+  = ListComplete
+  | ListTruncated
+  deriving (Eq, Show)
+
+data TeamMemberList
+  = TeamMemberList
+      { teamMembers :: [Teams.TeamMember],
+        teamMemberListType :: ListType
+      }
+  deriving (Eq, Show, Generic)
+
```

Comment:
The type in `galley-types` is boolean blind. I didn't change that because it's client API facing so I don't want to break things there. Or you meant change that type and change the serialisation to be a bool?

Reply:
Yes, I meant change the type but keep it a boolean in JSON.

Reply:
Yes, I thought about it just after your comment 😅 Good idea.

Reply:
https://github.com/wireapp/wire-server/commit/5ad36a68bb11a405529e3a17ad78feea56071045